crates/services/shared-sequencer/src/lib.rs

```diff
@@ -0,0 +1,269 @@
+//! Shared sequencer client
+
+#![deny(clippy::arithmetic_side_effects)]
+#![deny(clippy::cast_possible_truncation)]
+#![deny(unused_crate_dependencies)]
+#![deny(missing_docs)]
+
+use anyhow::anyhow;
+use cosmrs::{
+    tendermint::chain::Id,
+    tx::{
+        self,
+        Fee,
+        MessageExt,
+        SignDoc,
+        SignerInfo,
+    },
+    AccountId,
+    Coin,
+    Denom,
+};
+use error::PostBlobError;
+use fuel_sequencer_proto::protos::fuelsequencer::sequencing::v1::MsgPostBlob;
+use http_api::{
+    AccountMetadata,
+    TopicInfo,
+};
+use ports::Signer;
+use prost::Message;
+use tendermint_rpc::Client as _;
+
+// Re-exports
+pub use config::{
+    Config,
+    Endpoints,
+};
+pub use prost::bytes::Bytes;
+
+mod config;
+mod error;
+mod http_api;
+pub mod ports;
+pub mod service;
+
+/// Shared sequencer client
+pub struct Client {
+    endpoints: Endpoints,
+    topic: [u8; 32],
+    ss_chain_id: Id,
+    gas_price: u128,
+    coin_denom: Denom,
+    account_prefix: String,
+}
+
+impl Client {
+    /// Create a new shared sequencer client from config.
+    pub async fn new(endpoints: Endpoints, topic: [u8; 32]) -> anyhow::Result<Self> {
+        let coin_denom = http_api::coin_denom(&endpoints.blockchain_rest_api)
+            .await?
+            .parse()
+            .map_err(|e| anyhow::anyhow!("{e:?}"))?;
+        let account_prefix =
+            http_api::get_account_prefix(&endpoints.blockchain_rest_api).await?;
+        let ss_chain_id = http_api::chain_id(&endpoints.blockchain_rest_api)
+            .await?
+            .parse()
+            .map_err(|e| anyhow::anyhow!("{e:?}"))?;
+        let ss_config = http_api::config(&endpoints.blockchain_rest_api).await?;
+
+        let mut minimum_gas_price = ss_config.minimum_gas_price;
+
+        if let Some(index) = minimum_gas_price.find('.') {
+            minimum_gas_price.truncate(index);
+        }
+        let gas_price: u128 = minimum_gas_price.parse()?;
+        // Ceil the gas price to the next integer.
+        let gas_price = gas_price.saturating_add(1);
```

Comment:
Do we have a rationale for this piece of code. I was wondering:
1. why we truncate? (and how can we be sure that _dot_ will be used as decimal separator)?
2. why we ceil to the next integer?

Reply:
We need an integer for this. Yep, dot is the separator here. Ceil makes sure that the number is always at least large enough.